modules/articles.py

```diff
@@ -0,0 +1,59 @@
+import newspaper
+import re
+
+class article_content():
+
+    def __init__(self, article_object):
+        #Whenever a class object is made, the article is tokenized and the dictionary file is opened as a set.
+        self.tokenized_words = nltk.word_tokenize(article_object.text)
+        self.article_object = article_object
+        with open('words_alpha.txt') as word_file:
+            self.valid_words = set(word_file.read().split())
+        
+        self.misspells = 0  # Set initial misspells to 0.
+
+    def citation_check(self):
+        
+        #Something NLTK stop words. These often aren't real citations. / to remove local files, and ad removal from scraped link.
+        self.not_citations = ("facebook.com", "twitter.com", "#", "mailto", "plus.google", "advertising", "advertisement", "ad.", "/", "whatsapp", "quora", "nav.", "instagram")
+
+        self.article_html = self.article_object.html
+        http = [self.article_object.url] + re.findall(r'href="[^\"]+', self.article_html) #Regex code for finding links, eg href="http://www.bbc.com" returns http://www.bbc.com.
+        self.unique_citations = [] #Initialize unique urls to null.
+
+        #Find all cleaned urls (without http, https or www)
+        for http_url in http:
+            #Convert https to http, convert www to null, convert \ to /, and convert href to null
+            http_url = http_url.replace('https', 'http').replace('www.', '').replace('\\', '/').replace('href="','')
+            #Split string into two - http:// part and following URL part, then split again using  and use the 0th element to get the general source.
+            splitted = http_url.split("http://")[-1].split("/")
+            if splitted[0].strip().startswith(self.not_citations) is False and splitted[0].strip() not in self.unique_citations:
+                self.unique_citations.append(splitted[0])
+        
+        #split_http stores all the unique websites/citations that the article webpage has linked to.
+        
+        self.number_unique_citations = len(self.unique_citations) - 1  #Unique citations = Length of all unique website links minus itself
+
+        return self.number_unique_citations
+
+    def spell_check(self):
+        for word in self.tokenized_words: #For every word in article
+            if word[0].isupper()==False: #Ignore if first letter is uppercase, else
```

Comment:
Only necessary when the word is not the first word in a sentence.

Reply:
True! I will try to fix this using punctuations in tokenized_words.

Reply:
Do you have any suggestions regarding the problem that might arise with sentences starting with proper nouns?
Like

> John went to the school.
Here, John is not a misspelling, but if we don't ignore the first word in the sentence, it will be counted as a misspell.

>Stoodunts go to the school

Here Stoodunts is a misspelling but if we ignore the first word in the sentence, it will be ignored and we won't be able to count it.

I thought of something that counts the misspelling with the frequency of it. If it occurs many times, then it probably is a proper noun, but I'm not certain as to how to go about deciding what threshold it will be. It has to be dependent on the article text length I'm guessing, but haven't thought of much more.

Reply:
`nltk.corpus.names.words` would be a viable option